pkg/cached/cache_test.go

```diff
@@ -1054,3 +1055,61 @@ func TestListDAG(t *testing.T) {
 		t.Fatalf("expected etag = %v, got %v", want, result.Etag)
 	}
 }
+
+func randomString(length uint) string {
+	bytes := make([]byte, 6)
+	rand.Read(bytes)
+	return string(bytes)
+
+}
+
+func NewRandomSource() cached.Data[int64] {
+	return cached.NewStaticSource(func() cached.Result[int64] {
+		bytes := make([]byte, 6)
+		rand.Read(bytes)
+		return cached.NewResultOK(rand.Int63(), randomString(10))
+	})
+}
+
+func repeatedGet(data cached.Data[int64], end time.Time, wg *sync.WaitGroup) {
+	for time.Now().Before(end) {
+		_ = data.Get()
+	}
+	wg.Done()
+}
+
+func TestThreadSafe(t *testing.T) {
+	end := time.Now().Add(time.Second)
+	wg := sync.WaitGroup{}
+	static := NewRandomSource()
+	wg.Add(1)
+	go repeatedGet(static, end, &wg)
+	result := cached.NewResultOK(rand.Int63(), randomString(10))
+	wg.Add(1)
+	go repeatedGet(result, end, &wg)
+	replaceable := cached.Replaceable[int64]{}
+	replaceable.Replace(NewRandomSource())
+	wg.Add(1)
+	go repeatedGet(&replaceable, end, &wg)
+	wg.Add(1)
+	go func(r *cached.Replaceable[int64], end time.Time, wg *sync.WaitGroup) {
+		for time.Now().Before(end) {
+			r.Replace(NewRandomSource())
+		}
+		wg.Done()
+	}(&replaceable, end, &wg)
+	merger := cached.NewMerger(func(results map[string]cached.Result[int64]) cached.Result[int64] {
+		return results["one"]
+	}, map[string]cached.Data[int64]{
+		"one": NewRandomSource(),
```

Comment:
Why not append the two strings or something? A single map element is basically a transformer so we're not really testing the merger with multiple sources?